x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsSource.java

```diff
@@ -147,4 +182,46 @@ Exception getQueryParsingException() {
     Map<String, Object> getQuery() {
         return queryProvider.getQuery();
     }
+
+    public boolean isFieldExcluded(String path) {
+        if (sourceFiltering == null) {
+            return false;
+        }
+
+        // First we check in the excludes as they are applied last
+        for (String exclude : sourceFiltering.excludes()) {
+            if (patternMatchesFieldPath(exclude, path)) {
+                return true;
+            }
+        }
+
+        // Now we can check the includes
+
+        // Empty includes means no further exclusions
+        if (sourceFiltering.includes().length == 0) {
+            return false;
+        }
+
+        for (String include : sourceFiltering.includes()) {
+            if (patternMatchesFieldPath(include, path)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private static boolean patternMatchesFieldPath(String pattern, String path) {
+        if (Regex.isSimpleMatchPattern(pattern)) {
+            return Regex.simpleMatch(pattern, path);
+        } else {
+            if (pattern.equals(path)) {
```

Comment:
We could move this `equals` check before regex check as it is likely faster to execute.
Also, we could use `else if` instead of a nested `if` to reduce indentation.